Subject: tell firmware to relocate itself
From: pmac@au1.ibm.com
References: 427960

update RPA note in addnote to tell firmware to
relocate itself to make room for the downloaded ELF binary
there must be enough room for the TFTP download

Signed-off-by: Olaf Hering <olh@suse.de>

---
 arch/powerpc/boot/addnote.c     |   14 ++++++++------
 arch/powerpc/kernel/prom_init.c |   15 +++++++++------
 2 files changed, 17 insertions(+), 12 deletions(-)

--- a/arch/powerpc/boot/addnote.c
+++ b/arch/powerpc/boot/addnote.c
@@ -41,16 +41,18 @@ char rpaname[] = "IBM,RPA-Client-Config"
  * it looks at the splpar field at least.  So these values need to be
  * reasonable.
  */
-#define N_RPA_DESCR	8
+#define N_RPA_DESCR	10
 unsigned int rpanote[N_RPA_DESCR] = {
-	0,			/* lparaffinity */
-	64,			/* min_rmo_size */
+	1,			/* lparaffinity */
+	128,			/* min_rmo_size */
 	0,			/* min_rmo_percent */
-	40,			/* max_pft_size */
+	46,			/* max_pft_size */
 	1,			/* splpar */
 	-1,			/* min_load */
-	0,			/* new_mem_def */
-	1,			/* ignore_my_client_config */
+	1,			/* new_mem_def */
+	0,			/* ignore_my_client_config */
+	1,			/* large_page_ready */
+	0,			/* force_alpha_mode */
 };
 
 #define ROUNDUP(len)	(((len) + 3) & ~3)
--- a/arch/powerpc/kernel/prom_init.c
+++ b/arch/powerpc/kernel/prom_init.c
@@ -682,10 +682,10 @@ static unsigned char ibm_architecture_ve
 	W(0xffffffff),			/* virt_base */
 	W(0xffffffff),			/* virt_size */
 	W(0xffffffff),			/* load_base */
-	W(64),				/* 128MB min RMA */
+	W(128),				/* 128MB min RMA */
 	W(0xffffffff),			/* full client load */
 	0,				/* min RMA percentage of total RAM */
-	48,				/* max log_2(hash table size) */
+	46,				/* max log_2(hash table size) */
 
 	/* option vector 3: processor options supported */
 	3 - 2,				/* length */
@@ -737,6 +737,8 @@ static struct fake_elf {
 			u32	min_load;
 			u32	new_mem_def;
 			u32	ignore_me;
+			u32	large_page_ready;
+			u32	force_alpha_mode;
 		} rpadesc;
 	} rpanote;
 } fake_elf = {
@@ -781,13 +783,14 @@ static struct fake_elf {
 		.type = 0x12759999,
 		.name = "IBM,RPA-Client-Config",
 		.rpadesc = {
-			.lpar_affinity = 0,
-			.min_rmo_size = 64,	/* in megabytes */
+			.lpar_affinity = 1,
+			.min_rmo_size = 128,	/* in megabytes */
 			.min_rmo_percent = 0,
-			.max_pft_size = 48,	/* 2^48 bytes max PFT size */
+			.max_pft_size = 46,	/* 2^46 bytes max PFT size */
 			.splpar = 1,
 			.min_load = ~0U,
-			.new_mem_def = 0
+			.new_mem_def = 1,
+			.large_page_ready = 1
 		}
 	}
 };
